heron/api/src/java/com/twitter/heron/api/Config.java

```diff
@@ -452,6 +487,28 @@ public static void setTopologyExactlyOnceEnabled(Map<String, Object> conf, boole
     conf.put(Config.TOPOLOGY_EXACTLYONCE_ENABLED, String.valueOf(exactOnce));
   }
 
+  public static void setTopologyPackingAlgorithm(Map<String, Object> conf,
+                                                 PackingAlgorithmType type) {
```

Comment:
The packing type is just a Class. It needs to be extensible so users can implement their own so an enum won't work.
```
public static void setTopologyPackingClass(Map<String, Object> conf,
    Class<? extends IPacking> packingClass);
public static void setTopologyRepackingClass(Map<String, Object> conf,
    Class<? extends IRepacking> repackingClass);
```